kafka-client/src/main/java/dev/responsive/kafka/internal/stores/RemoteWindowOperations.java

```diff
@@ -53,7 +53,7 @@
 import org.apache.kafka.streams.state.WindowStoreIterator;
 import org.slf4j.Logger;
 
-public class SegmentedOperations implements WindowOperations {
+public class RemoteWindowOperations implements WindowOperations {
```

Comment:
nit: WDYT about calling this `TtlWindowOperations` since expiry is based on ttl for this, as opposed to the SegmentedWindowOperations where we expire in segments

Or even just "UnsegmentedWindowOperations" -- my point is that it feels a bit weird to call the two variants "Segmented" and "remote", just makes it seem like the Segmented variant isn't remote. Just a thought, I'll leave it up to you

Reply:
Hmm, I guess I thought that segmented operations were mostly encapsulated by the flush manager. The only references to segments in this class is in the factory logic for the respective stores. So I think we can use this class for RS3 as well, just with different factory logic (perhaps we could even factor out this logic, but I was punting on that for now). So `RemoteWindowOperations` would be used for both segmented and TTL based expiration. Does that make sense?